mqtt/mqtt.js

```diff
@@ -80,14 +80,22 @@ function startMqtt() {
     return net.createServer(aedes.handle).listen(config.mqtt_port);
 }
 
-nats.subscribe('channel.*', {'queue':'mqtts'}, function (msg) {
+function parseMqttTopic(topic) {
+    // allow channels/any-char-allowed/message with 0|+ block /any-char-allowed
+    // the text messages and the last subtopic can not terminate with /
+    return /^channels\/(.+?)\/messages(\/.+[^\/]$)*$/.exec(topic)
+}
+
+nats.subscribe('channel.>', {'queue':'mqtts'}, function (msg) {
     var m = message.RawMessage.decode(Buffer.from(msg)),
-        packet;
+        packet, channel;
     if (m && m.Protocol !== 'mqtt') {
+        channel = /^([\w\-]+)(\.(.+))?$/.exec(m.Channel)
```

Comment:
Again, is this `channel` or `channelParts`? Looks like there are two selections here.